CMakeLists.txt

```diff
@@ -139,7 +139,7 @@ option(FLB_COVERAGE            "Build with code-coverage"      No)
 option(FLB_JEMALLOC            "Build with Jemalloc support"   No)
 option(FLB_REGEX               "Build with Regex support"     Yes)
 option(FLB_UTF8_ENCODER        "Build with UTF8 encoding support" Yes)
-option(FLB_UNICODE_ENCODER     "Build with Unicode (UTF-16LE, UTF-16BE) encoding support" Yes)
+option(FLB_UNICODE_ENCODER     "Build with Unicode (UTF-16LE, UTF-16BE) encoding support" CMAKE_CXX_COMPILER)
```

Comment:
Hi, this should be good to be sent as different but a small PR.
This is because I have never thought to purify CXX contamination.
It should be great to be disabled when CXX toolchain is disabled like embedded environment.

Reply:
Hi, thank you for the in depth review, much appreciated. I opened a PR here. https://github.com/fluent/fluent-bit/pull/10277. For the other commits, I will first try to upstream what is needed.